.github/workflows/sync-package-periodically.yml

```diff
@@ -0,0 +1,45 @@
+name: Sync Package Metadata
+
+on:
+  schedule:
+    - cron: '0 0 * * 6'
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    
+    env:
+     Codeql.Enabled: true
+     Enable_SyncPackage: false
+     
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Restore NPM cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: ${{ runner.os }}-node-
+
+      - name: Sync Package metadata
+        run: scripts/syncMetadata.sh
+
+      - name: Rebuild and validate website
+        run: scripts/rebuild.sh
+
+      - name: Commit Results
+        run: |
+            git config --local user.email "actions@github.com"
+            git config --local user.name "GitHub Actions"
+            git add commit.txt
+            git commit -m 'Data Updated' || 'No changes to commit'
+            git push origin || echo "No changes to commit"
```

Comment:
Future hardening should separate regenerating the website (no write permissions) from pushing to origin (write permissions).